### PR TITLE
MES-3429 handle rekey upload failures

### PR DESCRIPTION
**Pull request by buzzub**
## Description and relevant Jira numbers
- rename the Rekey Uploaded page to Rekey Upload Outcome and show a different message if the uploaded test was detected to be a duplicate
- create an `EndRekey` action to reset the rekey reason page state
- remove unused actions from the rekey reason page
- add a temporary end rekey button to allow testing (MES-3441 implementing properly)
- include the HTTP error in the `SendCurrentTestFailure` action
- simplify the `sendCurrentTestEffect`
- simplify the rekey reason reducer and handle is duplicate upload checking
- add the default timeout to test submission

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

IMG_0123

### CODE REVIEW

**Review by rossfauldsbjss — APPROVED** [review]
looks good.

File: src/modules/tests/tests.actions.ts [hunk 1]
```diff
@@ -89,7 +90,7 @@ export class SendCurrentTestSuccess implements Action {
 
 export class SendCurrentTestFailure implements Action {
   readonly type = SEND_CURRENT_TEST_FAILURE;
-  constructor(public slotId: string) {}
+  constructor(public slotId: string, public error: HttpErrorResponse) { }
```
Comment (szabikr) [inline src/modules/tests/tests.actions.ts]: Do we need to send around the `slotId` considering that we are uploading the **current test**?

File: src/pages/rekey-reason/rekey-reason.reducer.ts [hunk 2]
```diff
@@ -1,47 +1,47 @@
...
       };
     case testActions.SEND_CURRENT_TEST_FAILURE:
       return {
         ...state,
         uploadStatus: {
-          isUploading: false,
-          hasUploadSucceeded: false,
+          ...initialState.uploadStatus,
           hasUploadFailed: true,
+          isDuplicate: action.error && action.error.status === HttpStatusCodes.CONFLICT,
```
Comment (szabikr) [inline src/pages/rekey-reason/rekey-reason.reducer.ts]: Would be better to send the evaluation of this expression as an action payload

**Review by szabikr — APPROVED** [review]
Great stuff! Pointed out two minor issues. What do you think about them?

File: src/modules/tests/tests.actions.ts (hunk 1 above)
Comment (buzzub) [reply src/modules/tests/tests.actions.ts]: Yeah, I would just need to ensure that in the test effects I get the current test. Hmmm. I will make this change.

File: src/pages/rekey-reason/rekey-reason.reducer.ts (hunk 2 above)
Comment (buzzub) [reply src/pages/rekey-reason/rekey-reason.reducer.ts]: Updated

**Pull request merged**